lib/status.js

```diff
@@ -307,16 +309,27 @@
 		};
 
 		checkTimeout = function() {
-			callback(new Error('Health check timed out (> ' + this.healthcheckTimeout + ' sec)'));
+			return callback(new Error('Health check timed out (> ' + this.healthcheckTimeout + ' sec)'));
 		};
+		
+		var parsedUrl = URL.parse(req.url),
+			urlParams = null;
+
+		if (parsedUrl.query) {
+			urlParams = querystring.parse(parsedUrl.query);
+		}
+
+		if (urlParams && urlParams.service_name && urlParams.service_name !== this.config.name) {
+			return callback(new Error('Service name mismatch'));
```

Comment:
not sure how usefull this might be, but what if we'd include both expected and real service names into error message?